tutorials/distributed-ml/torch-tutorial-GAN/train.py

```diff
@@ -0,0 +1,471 @@
+import os
+import argparse
+import torch
+import torch.nn as nn
+import torch.optim as optim
+import torchmetrics
+import torchvision
+from torchvision import datasets, transforms
+from itwinai.torch.trainer import TorchTrainer
+from itwinai.torch.config import TrainingConfiguration
+from itwinai.loggers import MLFlowLogger
+from typing import (
+    Optional, Dict, Union, Literal
+)
+from itwinai.loggers import Logger
+from itwinai.torch.type import Metric
+import matplotlib.pyplot as plt
+import numpy as np
+from itwinai.torch.distributed import (
+    DeepSpeedStrategy
+)
+
+REAL_LABEL = 1
+FAKE_LABEL = 0
+Z_DIM = 100
+G_HIDDEN = 64
+IMAGE_CHANNEL = 1
+D_HIDDEN = 64
+
+NOISE_DIM = 100
```

Comment:
these need to be moved to the trainer config and/or Generator/Discrimitator constructor